rpcs3/rpcs3qt/emu_settings.cpp

```diff
@@ -221,34 +221,50 @@ void emu_settings::EnhanceComboBox(QComboBox* combobox, emu_settings_type type,
 		}
 	}
 
-	const std::string selected = GetSetting(type);
-	const QString selected_q = qstr(selected);
-	int index = -1;
-
-	if (is_ranged)
-	{
-		index = combobox->findData(selected_q);
-	}
-	else
+	// Since the QComboBox has localised strings, we can't just findText / findData, so we need to manually iterate through it to find our index
+	auto find_index = [&](const QString& value)
 	{
 		for (int i = 0; i < combobox->count(); i++)
 		{
 			const QVariantList var_list = combobox->itemData(i).toList();
 			ASSERT(var_list.size() == 2 && var_list[0].canConvert<QString>());
 
-			if (selected_q == var_list[0].toString())
+			if (value == var_list[0].toString())
 			{
-				index = i;
-				break;
+				return i;
 			}
 		}
+		return -1;
+	};
+
+	const std::string selected = GetSetting(type);
+	const QString selected_q  = qstr(selected);
+	int index  = -1;
+
+	if (is_ranged)
+	{
+		index = combobox->findData(selected_q);
+	}
+	else
+	{
+		index = find_index(selected_q);
 	}
 
 	if (index == -1)
 	{
 		const std::string def = GetSettingDefault(type);
 		cfg_log.error("EnhanceComboBox '%s' tried to set an invalid value: %s. Setting to default: %s", cfg_adapter::get_setting_name(type), selected, def);
-		combobox->setCurrentIndex(combobox->findData(qstr(def)));
+
+		if (is_ranged)
+		{
+			index = combobox->findData(selected_q);
+		}
+		else
+		{
+			index = find_index(qstr(def));
+		}
+
+		combobox->setCurrentIndex(index);
```

Comment:
looking at the code now, you can just drop the else path and use this afterwards

Reply:
Use what afterwards?

Reply:
this line

Reply:
Index needs to be updated based on the default value.  And index is only updated in the else (for non is_range).

Reply:
combobox->setCurrentIndex(index); is currently executed in both the if and the else block.
So it can just be refactored to be called after the if/else.
And you need to rebase